### PR TITLE
web: codemod `web/diff` components to CSS modules

### PR DESCRIPTION
**Pull request by valerybugakov**
## Context

This PR results from the [global-css-to-css-modules](https://github.com/sourcegraph/codemod/tree/main/src/transforms/globalCssToCssModule) codemod application with some manual fixes, caused by the usage of global CSS classes tied to a React component in other parts of the codebase: e.g., `.sidebar` class used outside of the `<Sidebar />` component.

## Changes

- The stylesheets in `client/web/src/components/diff` are converted to CSS modules.

### CODE REVIEW

File: client/web/src/components/diff/DiffSplitHunk.test.tsx [hunk 1]
```diff
@@ -131,32 +132,26 @@ describe('DiffSplitHunk', () => {
...
+            expect(diffLine.children[1]).toHaveClass(lineStyles.lineActive)
+            expect(diffLine.children[3]).toHaveClass(lineStyles.lineActive)
         })
 
         it('add active class when the anchor is clicked on ADDED lines', () => {
             const diffLine = queries.getByTestId('anchor_R163')
             fireEvent.click(diffLine.children[2].children[0])
 
             expect(diffLine.children).toHaveLength(4)
-            expect(diffLine.children[0]).not.toHaveClass('diff-hunk--split__line--active-num')
```
Comment (5h1rU) [inline client/web/src/components/diff/DiffSplitHunk.test.tsx:145]: Is this deletion intentional?

File: client/web/src/components/diff/DiffSplitHunk.test.tsx [hunk 2]
```diff
@@ -131,32 +132,26 @@ describe('DiffSplitHunk', () => {
...
+            expect(diffLine.children[1]).not.toHaveClass(lineStyles.lineActive)
+            expect(diffLine.children[3]).toHaveClass(lineStyles.lineActive)
         })
 
         it('add active class when the anchor is clicked on DELETED lines', () => {
             const diffLine = queries.getByTestId('anchor_L164')
             fireEvent.click(diffLine.children[0].children[0])
 
             expect(diffLine.children).toHaveLength(4)
-            expect(diffLine.children[0]).toHaveClass('diff-hunk--split__line--active-num')
```
Comment (5h1rU) [inline client/web/src/components/diff/DiffSplitHunk.test.tsx:156]: do you need to remove these lines?

File: client/web/src/components/diff/Lines.tsx [hunk 3]
```diff
@@ -82,29 +81,25 @@ export const Line: React.FunctionComponent<Line> = ({
         <>
             {lineNumbers && (
                 <td
-                    className={classNames('diff-hunk__num', `${hunkStyles.hunkContent}-num`, `${className}-num`)}
```
Comment (5h1rU) [inline client/web/src/components/diff/Lines.tsx:85]: The suffix `-num` was removed. Is this correct?

File: client/web/src/components/diff/DiffBoundary.tsx [hunk 4]
```diff
@@ -16,10 +19,14 @@ interface DiffBoundaryContentProps extends DiffBoundaryProps {
...
-        <td className={classNames('diff-boundary__content', props.contentClassName)} data-diff-marker=" ">
+        {props.lineNumbers && <td className={styles.num} data-diff-boundary-num={true} colSpan={props.colspan} />}
+        <td
+            className={classNames(styles.content, props.contentClassName)}
+            data-diff-boundary-content={true}
+            data-diff-marker=" "
+        >
             {props.oldRange.lines !== undefined && props.newRange.lines !== undefined && (
-                <code className="diff-hunk__line--code diff-hunk__content">
+                <code className={classNames(diffHunkStyles.content)}>
```
Comment (5h1rU) [inline client/web/src/components/diff/DiffBoundary.tsx:29]: is this `classNames` necessary?

File: client/web/src/components/diff/Lines.tsx (hunk 3 above)
Comment (valerybugakov) [reply client/web/src/components/diff/Lines.tsx:85]: After inspecting styles more closely, I found that the `-num` suffix doesn't add any new styles compared to the class without it. Tested it locally and compared it to the production implementation. 

Let me know if I'm missing something.

File: client/web/src/components/diff/DiffSplitHunk.test.tsx (hunk 2 above)
Comment (valerybugakov) [reply client/web/src/components/diff/DiffSplitHunk.test.tsx:156]: Let's continue [here](https://github.com/sourcegraph/sourcegraph/pull/24660#discussion_r703636591). If the `-num` suffix is required, I'll add it back.

File: client/web/src/components/diff/DiffSplitHunk.test.tsx (hunk 1 above)
Comment (valerybugakov) [reply client/web/src/components/diff/DiffSplitHunk.test.tsx:145]: Let's continue [here](https://github.com/sourcegraph/sourcegraph/pull/24660#discussion_r703636591). If the `-num` suffix is required, I'll add it back.

**Review by 5h1rU — APPROVED** [review]

**Review by tjkandala — APPROVED** [review]
Nice, cool codemod!

**Pull request merged**